crates/cast/bin/cmd/send.rs

```diff
@@ -171,14 +194,54 @@ impl SendTxArgs {
 
             tx::validate_from_address(eth.wallet.from, from)?;
 
-            let (tx, _) = builder.build(&signer).await?;
-
-            let wallet = EthereumWallet::from(signer);
-            let provider = ProviderBuilder::<_, _, AnyNetwork>::default()
-                .wallet(wallet)
-                .on_provider(&provider);
-
-            cast_send(provider, tx, cast_async, confirmations, timeout, to_json).await
+            if zksync_params.zksync {
+                // ZkSync transaction
+                let paymaster_params = zksync_params
+                    .paymaster_address
+                    .and_then(|addr| zksync_params.paymaster_input.map(|input| (addr, input)))
+                    .map(|(addr, input)| PaymasterParams {
+                        paymaster: Address::from_str(&addr)
+                            .expect("Invalid paymaster address")
+                            .to_h160(),
+                        paymaster_input: Bytes::from_str(&input)
+                            .expect("Invalid paymaster input")
+                            .to_vec(),
+                    });
+
+                // Build EIP712 transaction for ZKSync
+                let tx = foundry_zksync_core::new_eip712_transaction(
+                    builder.build(&signer).await?.0,
+                    Vec::new(), // Empty factory_deps
+                    paymaster_params,
+                    &provider,
+                    signer,
+                )
+                .await
+                .map_err(|e| eyre::eyre!("Failed to create EIP712 transaction: {}", e))?;
+
+                // Use send_raw_transaction for ZKSync
+                let tx_hash = provider.send_raw_transaction(&tx).await?.tx_hash().to_owned();
+                let cast = Cast::new(provider);
+                handle_transaction_result(
+                    &cast,
+                    &tx_hash,
+                    cast_async,
+                    confirmations,
+                    timeout,
+                    to_json,
+                )
+                .await
```

Comment:
does it make sense to have this inside `cast_send_zk(..)` to be consistent instead of unwrapping it? A bit of duplication is fine as long as we have both `cast_send` and `cast_send_zk` living next to each other so we can verify the changes.